ion/binaryreader_test.go

```diff
@@ -16,13 +16,171 @@
 package ion
 
 import (
+	"bytes"
 	"fmt"
 	"math"
 	"math/big"
 	"testing"
 	"time"
 )
 
+const binaryFilePath = "testdata/binary/"
+
+func TestReadBinarySingleSymbol(t *testing.T) {
+	fileBytes := loadFile(t, binaryFilePath+"single_symbol.bindat")
+	r := NewReader(bytes.NewReader(fileBytes))
+	assertSingleSymbolTest(t, r)
+}
+
+func TestReadBinaryFieldNameSymbolAndSymbolValues(t *testing.T) {
+	table1 := NewSharedSymbolTable("table1", 1, []string{"s1", "s2"})
+	table2 := NewSharedSymbolTable("table2", 1, []string{"s3", "s4"})
+	catalog := NewCatalog(table1, table2)
+
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf, table1, table2)
```

Comment:
I generally tried to stay away from using the writers to test the readers and vice versa, lest there be a bug in both that round-trips data correctly but doesn't serialize it correctly per the spec. Could we generate this bit of test data once, make sure it looks correct, and then inline it?

Reply:
Removed this test and instead created `TestReadBinaryFieldNames` and `TestReadBinarySymbols` tests that takes in inline data